pkg/sql/logictest/testdata/logic_test/alter_column_type

```diff
@@ -161,3 +161,210 @@ ALTER TABLE t ALTER s TYPE STRING
+2016-01-26 16:10:10.55 +0100 CET
+2016-01-26 16:10:10.556 +0100 CET
+
+# ensure column families stay the same
+statement ok
+CREATE TABLE t7(id INT, id2 INT, FAMILY f1 (id), FAMILY f2 (id2));
+INSERT INTO t7 VALUES (1), (2), (3);
+ALTER TABLE t7 ALTER COLUMN id2 TYPE STRING;
+
+query TT
+SHOW CREATE TABLE t7
+----
+t7      CREATE TABLE t7 (
+              id INT8 NULL,
+              id2 STRING NULL,
+              FAMILY f1 (id, rowid),
+              FAMILY f2 (id2)
+            )
+
+# ensure default expressions are casted correctly.
+statement ok
+CREATE TABLE t8 (x INT DEFAULT 1, y INT);
+INSERT INTO t8 (y) VALUES (1), (2), (3);
+ALTER TABLE t8 ALTER COLUMN x TYPE DATE;
+INSERT INTO t8 (y) VALUES (4);
+
+query TI
+SELECT * FROM t8 ORDER BY y
+----
+1970-01-02 00:00:00 +0000 +0000  1
+1970-01-02 00:00:00 +0000 +0000  2
+1970-01-02 00:00:00 +0000 +0000  3
+1970-01-02 00:00:00 +0000 +0000  4
+
+# ensure a runtime error rollsback correctly.
+statement ok
+
+statement ok
+CREATE TABLE t9 (x INT PRIMARY KEY)
+
+statement error pq: unimplemented: alter column type requiring rewrite of on-disk data is currently not supported for columns that are part of an index
+ALTER TABLE t9 ALTER COLUMN x TYPE STRING
+
+statement ok
+CREATE TABLE t10 (x INT, y INT, INDEX(x, y))
+
+statement error pq: unimplemented: alter column type requiring rewrite of on-disk data is currently not supported for columns that are part of an index
+ALTER TABLE t10 ALTER COLUMN y TYPE STRING
+
+statement ok
+CREATE TABLE t11 (x INT)
+
+statement error pq: unimplemented: alter column type using expression is not supported\nHINT: You have attempted to use a feature that is not yet implemented.\nSee: https://github.com/cockroachdb/cockroach/issues/47706
+ALTER TABLE t11 ALTER COLUMN x TYPE STRING USING CAST(x AS STRING)
+
+# alter column type should not drop the column's owned sequence.
+statement ok
+CREATE TABLE t12 (x INT)
+
+statement ok
+CREATE SEQUENCE seq OWNED BY t12.x
+
+statement ok
+ALTER TABLE t12 ALTER COLUMN x TYPE STRING
+
+query TT
+SHOW CREATE SEQUENCE seq
+----
+  seq CREATE SEQUENCE seq MINVALUE 1 MAXVALUE 9223372036854775807 INCREMENT 1 START 1
+
+# dropping the column after the schema change should drop the sequence.
+
+statement ok
+ALTER TABLE t12 DROP COLUMN x
+
+statement error relation "seq" does not exist
+SHOW CREATE SEQUENCE seq
+
+# alter column type should maintain dependent sequences.
+statement ok
+CREATE SEQUENCE seq
+
+statement ok
+CREATE TABLE t13 (x INT, y INT DEFAULT nextval('seq'))
+
+statement ok
+ALTER TABLE t13 ALTER COLUMN y TYPE STRING
+
+statement error pq: cannot drop sequence seq because other objects depend on it
+DROP SEQUENCE seq
+
+# column types cannot be altered in the same txn the table is created.
+
+statement error pq: unimplemented: performing an ALTER TABLE ... ALTER COLUMN TYPE in the same txn as the table was created is not supported
+BEGIN;
+CREATE TABLE t14 (x INT);
+ALTER TABLE t14 ALTER COLUMN x TYPE STRING;
+COMMIT;
```

Comment:
The file is empty, so I'm adding a comment here -- what is the alter_column_type_other file?